LOCAL_DEV.md

```diff
@@ -0,0 +1,234 @@
+# 😺 Kitty Items - Local Development 👩‍💻
+
+## 👋 Introduction
+We chose to connect Kitty Items to Flow's testnet because we wanted users to see how easy it was to take a non-trivial application to a live Flow network.
+
+Of course, some users may want to expreiment with their smart contracts before deploying them, or otherwise prototype their appliccation before publishing it.
+In this section we'll detail how to set up your local environment for dapp development on Flow.
+
+## 🛠 Tools you'll use
+Local development on Flow is made possible using the following tools.
+
+- The `flow-cli`
+  - We'll use the Flow command-line interface (`flow-cli`) to create accounts and bootstrap our contracts onto the Flow emulator.
+  - `flow-cli` [documentation](https://docs.onflow.org/flow-cli)
+- The Flow Emulator
+  - The Flow Emulator is a Flow blockchain that you can run locally. It has all the features and functionality of the live network because it uses the same software to process transactions and execute code as Flow's execution nodes. (Note: At the moment the emulator can't / does not reflect the state of the live network)
+  - Flow emulator [documentation](https://docs.onflow.org/emulator)
+- The `fcl-dev-wallet`
+  - The dev wallet uses the same protocols required by `fcl` and Flow that are being used in production in consumer Flow wallets, such as Blocto.
+  - `fcl-dev-wallet` [documentation](https://github.com/onflow/fcl-dev-wallet)
+
+## ✨ Getting Started
+
+### 1. Install the Flow CLI
+
+Before you start, install the [Flow command-line interface (CLI)](https://docs.onflow.org/flow-cli).
+
+_⚠️ This project requires `flow-cli v0.15.0` or above._
+
+### 2. Clone the project
+
+```sh
+git clone https://github.com/onflow/kitty-items.git
+```
+
+### 2.b Initialize the `fcl-dev-wallet` submodule
+
+If you've already cloned the project, or you're starting fresh, you'll neeed to include the `fcl-dev-wallet`.
+
+```sh
+git submidule init && git submodule update
+```
+
+This command will clone the `fcl-dev-wallet` as a submodule in your repository. We're including the code as a submodule (and not a package) for the time being, because the wallet is still in early development.
+To ensure you have the latest version of the wallet, you can simply run `git submodule update` to fetch the latest code.
+
+### 3. Create `.env` files
+
+Create `.env.local` in the `web` and `api` folders.
```

Comment:
Could this step be omitted by just having a `.env.local` file already committed to GitHub, but then after it is included in the repo add `.env.local` to `.gitignore`?

Reply:
It's better that it's in the `.gitignore` first as a safeguard, in case folks forget to add it, and send private keys to their repos.

Reply:
I'm not saying to remove it from .gitignore, what I'm saying is that it could be added to repo as `.env` and after it is included in repo the `.env` could be added to .gitignore this way you already have the file so no need copying but it won't be submitted by mistake to repo by anyone.

Reply:
Yeah this `.env` shenanigans sucks.

Reply:
I found this tool which is saving some headaches reading environment variables into the app. 
Allowed me to remove the env switching logic `require('dotenv').config()` nonsense.

Reply:
https://www.npmjs.com/package/env-cmd

Reply:
Is there a way we could avoid/simplify this without adding too much overhead? I haven't given it too ,much thought until now?

Reply:
@psiemens @srinjoyc @sideninja @MaxStalker 
pew pew pew can we fix this?

Reply:
https://12factor.net/config